http/src/lib.rs

```diff
@@ -27,6 +30,14 @@ pub async fn http_server(
     Ok(Server::new(HttpRawServer::bind(addr).await?))
 }
 
+/// Starts a [`Server`](../Server) object that serves HTTP with a whitlist access control.
+pub async fn http_server_with_acl(
+    addr: &SocketAddr,
+    allowed_hosts: Option<Vec<Host>>,
```

Comment:
I'd be more in favour of using an enum for this all the way down, as I found the `Option` type very confusing in the past, like:
```
allowed_hosts: None // - that means that all hosts are allowed, in contrary to what you read
allowed_hosts: Some(vec![]) // - this means that no hosts are allowed
```

Reply:
Something like this: 
`
pub enum AccessControl {
All, 
Restricted(Vec<Hosts>),
}
`
?

Reply:
```Rust
/// Access control for http header 'host' 
#[derive(Clone)]
pub enum AllowedHosts {
    /// Allow requests from any host 
    Any,
    /// Allow only a selection of specific hosts
    Only(Vec<Host>),
}
```